tests/unit/notification/QueueProcessor.spec.js

```diff
@@ -0,0 +1,151 @@
+            );
+            done();
+        });
+    });
+
+    // FIXME: this test could fit better in Integration as it would
+    // also test the resilience of the producer, but at the time this
+    // test was written Bucket Notification tests are not yet running
+    // in Integration (refs: BB-419, INTGR-895)
+    it('should process 10K notifications with 1K concurrency to send to external destination',
+    function test10KNotif(done) {
+        this.timeout(5000);
+
+        const origLogger = qp.logger;
+        qp.logger = {
+            debug: () => {},
+            info: () => {},
+        };
+        // fake how Kafka Producer delivers pending delivery reports at a regular interval
+        let pendingDeliveryReports = [];
+        const flushDeliveryReportsInterval = setInterval(() => {
+            const _pendingDeliveryReports = pendingDeliveryReports;
+            pendingDeliveryReports = [];
+            for (const deliveryReportCb of _pendingDeliveryReports) {
+                deliveryReportCb();
+            }
+        }, 200);
+        const send = sinon.stub().callsFake((messages, cb) => {
+            // push the delivery report to be delivered asynchronously later
+            pendingDeliveryReports.push(cb);
+        });
+        qp._destination._notificationProducer = {
+            send,
+        };
+        async.timesLimit(10000, 1000, (i, iDone) => qp.processKafkaEntry({
+            value: JSON.stringify({
+                bucket: 'mybucket',
+                key: `key-${i}`,
+                eventType: 's3:ObjectCreated:Put',
+                value: '{}',
+            }),
+        }, iDone), err => {
+            assert.ifError(err);
+            assert.strictEqual(pendingDeliveryReports.length, 0);
+            qp.logger = origLogger;
+            clearInterval(flushDeliveryReportsInterval);
+            done();
+        });
+    });
+
+    it('should send notification to external destination with delivery error', done => {
+        const send = sinon.stub().callsFake((messages, cb) => cb(new Error('delivery error')));
+        qp._destination._notificationProducer = {
+            send,
+        };
+        qp.processKafkaEntry({
+            value: JSON.stringify({
+                bucket: 'mybucket',
+                key: 'key',
+                eventType: 's3:ObjectCreated:Put',
+                value: '{}',
+            }),
+        }, err => {
+            assert.ifError(err);
+
+            assert(send.calledOnce);
+            setTimeout(done, 100);
+        });
+    });
+
+    it('should not send an entry without "eventType" attribute', done => {
+        const send = sinon.spy();
+        qp._destination._notificationProducer = {
+            send,
+        };
+        qp.processKafkaEntry({
+            value: JSON.stringify({
+                bucket: 'mybucket',
+                key: 'key',
+                // no "eventType"
+                value: '{}',
+            }),
+        }, err => {
```

Comment:
if there's no eventType, is there an error/log message or it just silently doesn't send?

Reply:
I think this is where the check happens: https://github.com/scality/backbeat/blob/development/8.6/extensions/notification/utils/config.js#L85

I believe there is no particular log.

For the context, I closed a previous iteration of this PR (https://github.com/scality/backbeat/pull/2421) that was not waiting for the callback of `send` but we decided to rework it differently, the point of this test was then mostly to trigger a case where a message is not sent for regression testing, not particularly to check all details of the particular case where `eventType` is not passed. I kept it because I think it has some value to test this case but it's not anymore directly related to the change.